File: sdks/ios/Sources/XMTPiOS/Codecs/DeletedMessage.swift
import Foundation

/// Content type ID for deleted messages (used in enriched messages).
public let ContentTypeDeletedMessage = ContentTypeID(
	authorityID: "xmtp.org",
	typeID: "deletedMessage",
	versionMajor: 1,
	versionMinor: 0,
)

/// Represents a message that has been deleted.
/// This is used in enriched messages to indicate that a message was deleted.
public struct DeletedMessage: Codable, Equatable {
	public let deletedBy: DeletedBy

	public init(deletedBy: DeletedBy) {
		self.deletedBy = deletedBy
	}
}

/// Indicates who deleted the message.
public enum DeletedBy: Codable, Equatable {
	/// The original sender deleted their own message
	case sender
	/// An admin deleted the message
	case admin(inboxId: String)
}

File: sdks/android/shared/src/main/java/org/xmtp/kotlin/codecs/DeletedMessage.kt
package org.xmtp.kotlin.codecs

import org.xmtp.kotlin.InboxId

/**
 * Represents a message that has been deleted.
 */
data class DeletedMessage(
    val deletedBy: DeletedBy,
)

/**
 * Indicates who deleted the message.
 */
sealed class DeletedBy {
    /** The original sender deleted their own message */
    object Sender : DeletedBy()

    /** An admin deleted the message */
    data class Admin(
        val inboxId: InboxId,
    ) : DeletedBy()
}
